test/io/jesi/backpack/spy_test.cljc

```diff
@@ -23,34 +28,53 @@
          :cljs (is (= '(clojure.core/when js/goog.DEBUG (prn "hello"))
                       (macroexpand-1 '(io.jesi.backpack.spy/when-debug (prn "hello")))))))))
 
+(def file #?(:cljs "io.jesi.backpack.spy-test"
+             :clj  *file*))
+
+(def line (atom nil))
+
+(defn- set-line [n]
+  (reset! line n))
+
+(defn- add-line [n]
+  (swap! line + n))
+
 (deftest prn-test
 
   (testing "spy/prn"
 
     #?(:clj (testing "is a macro"
               (bp/macro? `spy/prn)))
 
-    (testing "expands to println"
-      (is (= '(io.jesi.backpack.spy/when-debug (clojure.core/println "a:" (clojure.core/pr-str a) "b:" (clojure.core/pr-str b)))
-             (macroexpand-1 '(io.jesi.backpack.spy/prn a b)))))
-
     (testing "prns"
-      #?(:clj  (do
-
-                 (testing "the specified values"
-                   (is (= "a: 1\n" (with-out-str (spy/prn a))))
-                   (is (= "a: 1 b: 2\n" (with-out-str (spy/prn a b)))))
-
-                 (testing "literal expressions"
-                   (is (= "1: 1\n" (with-out-str (spy/prn 1))))
-                   (is (= "a: \"a\"\n" (with-out-str (spy/prn "a"))))
-                   (is (= "(inc 1): 2\n" (with-out-str (spy/prn (inc 1)))))
-                   (is (= "((comp inc dec) 1): 1\n" (with-out-str (spy/prn ((comp inc dec) 1)))))))
-
-         :cljs (testing "nothing when debug"
-                 (is (false? js/goog.DEBUG))
-                 (is (= "" (with-out-str (spy/prn a))))
-                 (is (= "" (with-out-str (spy/prn a b)))))))))
+      (spy/with-spy
+        #?(:cljs (set! js/goog.DEBUG true))
```

Comment:
Could you make this set! a private function above instead of doing conditionals each time?

```clojure
(defn- set-debug [v] 
    #?(:cljs (set! js/goog.DEBUG v)))
...
...
(spy/with-spy
    (set-debug true)
    ....)
```